src/api/Main/BuisnessLayer/Shrooms.Domain/Services/UserService/UserService.cs

```diff
@@ -150,10 +153,30 @@ public void Delete(string userToDelete, UserAndOrganizationDTO userOrg)
             UnassignUserFromWalls(userToDelete, userOrg.OrganizationId);
             _userManager.RemoveLogins(userToDelete);
 
+            await Anonymize(user, userOrg);
+
             _usersDbSet.Remove(user);
             _uow.SaveChanges(userOrg.UserId);
         }
 
+        private async Task Anonymize(ApplicationUser user, UserAndOrganizationDTO userOrg)
+        {
+            await _pictureService.RemoveImage(user.PictureId, userOrg.OrganizationId);
+
+            user.Email = string.Empty;
+            user.FirstName = string.Empty;
+            user.LastName = string.Empty;
+            user.Bio = string.Empty;
+            user.PhoneNumber = string.Empty;
+            user.UserName = string.Empty;
```

Comment:
Are you sure it will conform all constraints? Like "unique" constraint, "not null/empty", etc?

Reply:
Yes, you are right. There was a constraint on _Email_ field. I've fixed that not.

Reply:
> Yes, you are right. There was a constraint on _Email_ field. I've fixed that now
> .

Only email? What about UserName?

Reply:
Only email? What about UserName?

Reply:
Yes, only email.

Reply:
I still would suggest to generate some random string for all fields with not-null constraint in DB. Or at least for such important fields like _username_, emails (all emails), etc. In other words - anonymization shouldn't break the user, it just should make it impossible to identify that user.

Reply:
Only _username_ field doesn't allow nulls. I'll fix this then. I could generate 1 Guid and set it to all different fields, what do you think?

Reply:
Should be fine, I think.